pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -18,8 +18,8 @@
 from os.path import join as pjoin
 
 from pymedphys._dicom.anonymise import (
-    anonymise_directory,
-    anonymise_file,
+    _anonymise_directory,
```

Comment:
this is a bit disturbing, but maybe because of my having both an OO background and a .NET background
a namespace that is private, i.e. pymedphys._dicom suggests that it should only be used by someone who is programming within the python package.
But a function that is private, i.e. _anonymise_directory, suggests that it should only be used within the module/file, or at most, within the same namespace (files that are children of the same directory), and in tests of that module (although there is a philosophy that says unit tests should be of public methods/functions and not private methods/functions... not that I adhere to that but I respect the idea that a refactoring should not break unit tests, and altering private methods should not be considered a break in contract or a change in requirements.

I'm willing to go along but I'd like to see anonymise_directory and anonymise_file made part of the internal (public to a private module) API of _dicom.anonymise

Reply:
Yup, I agree. I don't like what's happened here either. `_anonymise_directory` should mean internally that it should be kept within that specific namespace and not used across the library.

> I'm willing to go along but I'd like to see anonymise_directory and anonymise_file made part of the internal (public to a private module) API of _dicom.anonymise

Honestly let's close this PR. It was a quick and flick attempt to try and refactor... but it was done poorly. I'll close this one. But did you want to do something similar so that we can get the renaming and API facade out of the way as a refactor that doesn't actually change any logic. And then we can dive into the actual changes in the other PR.

Reply:
I got started in splitting things up as you asked (core.py, api.py, strategies.py, and __init__.py), but it's not trivial because of dependencies

Reply:
I'm happy to give it a shot of you'd like? Probably won't get a chance till tonight though. Sorry I jumped in here. I was trying to be helpful... But I was probably more of a hindrance...